homeassistant/components/switchbot_cloud/__init__.py

```diff
@@ -0,0 +1,76 @@
+"""The SwitchBot via API integration."""
+from asyncio import gather
+from dataclasses import dataclass
+from logging import getLogger
+
+from switchbot_api import CannotConnect, Device, Remote, SwitchBotAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_API_TOKEN, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import SwitchBotCoordinator
+
+_LOGGER = getLogger(__name__)
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+@dataclass
+class SwitchbotDevices:
+    """Switchbot devices data."""
+
+    switches: list[Device | Remote]
+
+
+@dataclass
+class SwitchbotCloudData:
+    """Data to use in platforms."""
+
+    api: SwitchBotAPI
+    devices: SwitchbotDevices
+
+
+async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry) -> bool:
+    """Set up SwitchBot via API from a config entry."""
+    token = config.data[CONF_API_TOKEN]
+    secret = config.data[CONF_API_KEY]
+
+    api = SwitchBotAPI(token=token, secret=secret)
+    try:
+        devices = await api.list_devices()
+    except CannotConnect as ex:
+        raise ConfigEntryNotReady from ex
```

Comment:
We don't want to keep retrying when we know auth is invalid.

Reply:
My bad, didn't know this. Thanks

Reply:
Its kinda a rare case for new integrations first PRs since most of the time we `raise ConfigEntryAuthFailed` but we generally don't implement reauth in the first PR since it makes reviews longer and the goal is minimum viable for first merge to reduce frustration with the code review process